cl/opinion_page/views.py

```diff
@@ -293,8 +294,30 @@ def view_opinion(request, pk, _):
             'caller': 'view_opinion',
         }
         citing_clusters = conn.raw_query(**q).execute()
+
+        # Related opinions with Solr-MoreLikeThis query
+        # (Beta test: feature is only available for specific user groups)
+        if request.user.is_authenticated and (request.user.is_superuser or request.user.is_staff
+            or (hasattr(settings, 'RELATED_USER_GROUPS') and
+                request.user.groups.filter(name__in=settings.RELATED_USER_GROUPS).exists())):
```

Comment:
This seems overly complicated, no? Let's just create a group, and if the user is in it, show them the things? Why think about staff, auth, settings, etc?

Reply:
It checks all the other conditions to not query the database all the time. The settings check is added for those cases when the settings variable does not exist (is deleted by accident..).